processor/tailsampling/factory.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tailsampling
+
+import (
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-service/config/configerror"
+	"github.com/open-telemetry/opentelemetry-service/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-service/consumer"
+	"github.com/open-telemetry/opentelemetry-service/processor"
+	"github.com/open-telemetry/opentelemetry-service/service/builder"
+)
+
+const (
+	// The value of "type" Tail Sampling in configuration.
+	typeStr = "tail-sampling"
+)
+
+// Factory is the factory for Tail Sampling processor.
+type Factory struct {
+}
+
+// Type gets the type of the config created by this factory.
+func (f *Factory) Type() string {
+	return typeStr
+}
+
+// CreateDefaultConfig creates the default configuration for processor.
+func (f *Factory) CreateDefaultConfig() configmodels.Processor {
+	return &builder.TailBasedCfg{
+		DecisionWait: 30 * time.Second,
+		NumTraces:    50000,
```

Comment:
Is there a reason for those numbers?

Reply:
I'm not sure either. This is copied from https://github.com/open-telemetry/opentelemetry-service/blob/4c7c3ca57110c32803b56cec3fe4733afd74c952/service/builder/sampling_builder.go#L170-L176
@pjanotti any idea on those numbers?

Reply:
These are arbitrary and there is no good criteria other than "reasonableness", that I am aware :), to select those defaults.